applications/callflow/src/cf_endpoint.erl

```diff
@@ -1216,19 +1216,33 @@ generate_sip_headers(Endpoint, Call) ->
 generate_sip_headers(Endpoint, Acc, Call) ->
     Inception = whapps_call:inception(Call),
 
-    HeaderFuns = [fun maybe_add_sip_headers/1
+    HeaderFuns = [fun(J) -> maybe_add_sip_headers(J, Endpoint, Call) end
                   ,fun(J) -> maybe_add_alert_info(J, Endpoint, Inception) end
                   ,fun(J) -> maybe_add_aor(J, Endpoint, Call) end
                  ],
     lists:foldr(fun(F, JObj) -> F(JObj) end, Acc, HeaderFuns).
 
--spec maybe_add_sip_headers(wh_json:object()) -> wh_json:object().
-maybe_add_sip_headers(JObj) ->
-    case kz_device:custom_sip_headers(JObj) of
-        'undefined' -> JObj;
-        CustomHeaders -> wh_json:merge_jobjs(CustomHeaders, JObj)
+-spec maybe_add_sip_headers(wh_json:object(), wh_json:object(), whapps_call:call()) -> wh_json:object().
+maybe_add_sip_headers(JObj, Endpoint, Call) ->
+    case ?MODULE:get(Call) of
+        {'ok', AuthorizingEndpoint} ->
+            MergeFuns = [fun() -> kz_device:custom_sip_headers(Endpoint) end
+                         ,fun() -> kz_device:custom_sip_headers_inbound(Endpoint) end
+                         ,fun() -> kz_device:custom_sip_headers_outbound(AuthorizingEndpoint) end
+                        ],
```

Comment:
@icehess i think it would be cleaner if you search/replace every call to kz_device:custom_sip_headers to kz_device:custom_sip_headers_inbound, and have kz_device:custom_sip_headers_inbound handle both legacy and new format.